### ISSUE TITLE
`HyperparameterTuner` not keeping estimator environment variables

### ISSUE THREAD

**Issue by Guillem96** [aws/sagemaker-python-sdk#3598]
**Describe the bug**
When instanciating an Hyperparameter Tuning Job with `HyperparameterTuner` object, the environment variables declared in the underlying estimator are not provided to the Training Jobs.

When I declare a custom environment variable (`CUSTOM_ENV_VARIABLE`) in the estimator environemnt I should be able to access it in the created training jobs, but that's not the case.

When running the training jobs (of the HPO) I'm getting this output:

```
  | 2023-01-18T08:40:41.013+01:00 | ********************************************************************************
-- | -- | --
  | 2023-01-18T08:40:41.013+01:00 | Environment value:
  | 2023-01-18T08:40:41.013+01:00 | <NOT-DEFINED>
  | 2023-01-18T08:40:41.013+01:00 | ********************************************************************************
```

Saying that my variable is not defined (see the *to reproduce* section below).

**To reproduce**

```python
# Minimal estimator code
import os

env_value = os.environ.get("CUSTOM_ENV_VARIABLE", "<NOT-DEFINED>")

print("*" * 80)
print("Environment value:", env_value)
print("*" * 80)

# TBD: Training logic
# ...

print("Accuracy: 0.45345")  # so HPO does not complain
```

```python
# Code to instanciate the HPO job

import sagemaker
from sagemaker.pytorch import PyTorch
from sagemaker.tuner import HyperparameterTuner
from sagemaker.parameter import CategoricalParameter

pytorch_estimator = PyTorch(
    "minimal.py",
    instance_type="ml.m5.large",
    instance_count=1,
    framework_version="1.12.1",
    py_version="py38",
    environment={"CUSTOM_ENV_VARIABLE": "CUSTOM_VALUE"},
    metric_definitions=[
        {"Name": "accuracy", "Regex": r"^Accuracy: (\d*[.]?\d+)$"},
    ],
    role="<role_arn>",
)

tuner = HyperparameterTuner(
    pytorch_estimator,
    objective_metric_name="accuracy",
    hyperparameter_ranges={
        "mock": CategoricalParameter(["a", "b"]),
    },
    metric_definitions=[
        {"Name": "accuracy", "Regex": r"^Accuracy: (\d*[.]?\d+)$"},
    ],
    base_tuning_job_name="minimal-missing-env-bug",
)

tuner.fit({"train": "<s3_path>"})
```

**Expected behavior**

I should get this output:

```
 | 2023-01-18T08:40:41.013+01:00 | ********************************************************************************
-- | -- | --
  | 2023-01-18T08:40:41.013+01:00 | Environment value:
  | 2023-01-18T08:40:41.013+01:00 | CUSTOM_VALUE
  | 2023-01-18T08:40:41.013+01:00 | ********************************************************************************
```

Stating that the `CUSTOM_ENV_VARIABLE` is defined and has the `CUSTOM_VALUE` value.

**System information**

- **SageMaker Python SDK version**:  2.128.0
- **Framework name (eg. PyTorch) or algorithm (eg. KMeans)**: Pytorch
- **Framework version**: 1.12
- **Python version**: py38
- **CPU or GPU**: CPU and GPU (error happening in both machine types)
- **Custom Docker image (Y/N)**: N

**Comment by lorenzwalthert** [comment]
To me, this seems a duplicate of #2618, hopefully this will be closed with #3610 soon, since the underlaying functionality got [announced yesterday](https://aws.amazon.com/about-aws/whats-new/2023/01/amazon-sagemaker-automatic-model-tuning-environment-variables-tuning-jobs/).

**Comment by Guillem96** [comment]
@lorenzwalthert thank you for the update, I'll close the issue and startwatching the one you mention.
